onnxruntime/core/providers/cpu/rnn/deep_cpu_gru.cc

```diff
@@ -620,15 +620,19 @@ void UniDirectionalGru<T>::Compute(const gsl::span<const T>& inputs_arg,
 
     if (linear_before_reset_) {
       // copy Rbh to linear output
-      gsl::copy(batched_bias_Rh_.subspan(batched_bias_Rh_local - batched_bias_Rh_.begin(), batched_bias_Rh_local_end - batched_bias_Rh_local), linear_output_);
+      if (use_bias_) {
+        gsl::copy(batched_bias_Rh_.subspan(batched_bias_Rh_local - batched_bias_Rh_.begin(),
+                                           batched_bias_Rh_local_end - batched_bias_Rh_local),
+                  linear_output_);
+      }
 
       // compute Ht-1 * (Rh^T) + Rbh
       ComputeGemm(batch_size_, hidden_size_, hidden_size_, alpha,
```

Comment:
>ComputeGemm [](start = 6, length = 11)

Doesn't this also need to be inside the `if (use_bias_)` clause?

Reply:
I don't think so. It's doing the Ht-1 * Rh^T. It's just that if we're not adding the bias (Rbh) we don't need to copy that first. 

---
In reply to: [419267844](https://github.com/microsoft/onnxruntime/pull/3797#discussion_r419267844) [](ancestors = 419267844)